addons/project/models/project.py

```diff
@@ -1769,9 +1769,14 @@ def write(self, vals):
             if 'kanban_state' not in vals:
                 vals['kanban_state'] = 'normal'
         # user_ids change: update date_assign
-        if vals.get('user_ids') and 'date_assign' not in vals:
-            vals['date_assign'] = now
-
+        if not vals.get('user_ids', []):
```

Comment:
I think we don't cover a case with this changes. 🤔

If the user doesn't change the users assigned to the task then the `user_ids` won't be into the `vals` and so this condition will be satisfied. 🙂 

Also, we perhaps need to check if the `date_assign` is not manually set by the user to avoid replacing it by another value (`False` and `datetime.now`).

Reply:
As date_assign is a readonly field, I guess users cannot manually set a value :) I'll update the first part of your comment though, thanks!